features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js

```diff
@@ -122,6 +123,21 @@ class APIClient {
             if (data.headers.etag) {
                 APIClient.addETag(data.url, data.headers.etag);
             }
+
+            // If an unauthenticated response is received, we check whether the token is valid by introspecting it.
+            // If it is not valid, we need to clear the stored tokens (in cookies etc) in the browser by redirecting the
+            //   user to logout.
+            if (data.status === 401 && data.obj != null && data.obj.description === 'Unauthenticated request') {
+                const userData = AuthManager.getUserFromToken();
+                userData.then((user) => {
+                    if (user == null) {
+                        window.location = Configurations.app.context + Utils.CONST.LOGOUT_CALLBACK;
+                    }
+                }).catch((error) => {
+                    console.error('Error occurred while checking token status. Hence redirecting to login', error);
+                    window.location = Configurations.app.context + Utils.CONST.LOGOUT_CALLBACK;
+                });
```

Comment:
If we redirect user to 

```javascript
window.location = Configurations.app.context + Utils.CONST.LOGOUT_CALLBACK;
```

regardless of the promise susses or catch, we could put the above code in `.finally` callback